### PR TITLE
Fixed Numerical Stability Issues

### PR DESCRIPTION
**Pull request by cheng-chi**
Hi Tony! Thank you for this great implementation of MDN. 
However, I have encountered quite a few numerical issues while using it (gradient goes to NaN, invalid parameter etc), so I implemented some fixes:
1. To enforce positive standard deviation, I used ELU + 1 + epsilon instead of exponential.
2. When calculating loss for Pi distribution, use logits directly instead of torch.log(probs).

### CODE REVIEW

**Comment by tonyduan** [comment]
Terrific changes -- thank you for the PR!
